stargate/pkg/client/client.go

```diff
@@ -0,0 +1,209 @@
+package client
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/anypb"
+	"google.golang.org/protobuf/types/known/wrapperspb"
+
+	"github.com/stargate/stargate-grpc-go-client/stargate/pkg/auth"
+	pb "github.com/stargate/stargate-grpc-go-client/stargate/pkg/proto"
+)
+
+type StargateClient struct {
+	client       pb.StargateClient
+	conn         *grpc.ClientConn
+	authProvider auth.AuthProviderIFace
+}
+
+type Parameters struct {
+	Keyspace          string
+	Consistency       Consistency
+	PageSize          int32
+	PagingState       []byte
+	Tracing           bool
+	SkipMetadata      bool
+	Timestamp         int64
+	SerialConsistency SerialConsistency
+	NowInSeconds      int32
+}
+
+func NewStargateClient(target string, authProvider auth.AuthProviderIFace) (*StargateClient, error) {
+	conn, err := grpc.Dial(target, grpc.WithInsecure(), grpc.WithBlock())
+	if err != nil {
+		log.WithError(err).Error("Failed to create client")
+		return nil, fmt.Errorf("failed to create client: %v", err)
+	}
+
+	client := pb.NewStargateClient(conn)
+
+	return &StargateClient{
+		client:       client,
+		conn:         conn,
+		authProvider: authProvider,
+	}, nil
+}
+
+type Batch struct{}
+
+func NewQuery() *Query {
+	return &Query{
+		Parameters: Parameters{
+			Consistency:       UNSET,
+			SerialConsistency: UNSET_SERIAL,
+		},
+	}
+}
+
+func (s *StargateClient) ExecuteQuery(query *Query) (*Response, error) {
+	// TODO: [doug] configurable timeout?
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second+10)
```

Comment:
I think we should have variants of these functions that take a `Context` so that it can be chained with an existing application's context handling. wdyt?

Reply:
Sure